pkg/apis/sriovnetwork/v1/helper.go

```diff
@@ -97,8 +84,10 @@ func (s *SriovNetworkNicSelector) Selected(iface *InterfaceExt) bool {
 	if s.Vendor != "" && s.Vendor != iface.Vendor {
 		return false
 	}
-	if s.DeviceID != "" && s.DeviceID != iface.DeviceID {
-		return false
+	if s.DeviceID != "" {
+		if ((iface.NumVfs == 0 && s.DeviceID != iface.DeviceID) || (iface.NumVfs > 0 && s.DeviceID != SriovPfVfMap[iface.DeviceID])) {
```

Comment:
Will this prevent user from resetting vf to 0 for a PF device?

Reply:
No, it will not. The function just help to make sure that only the NodeState CR which contains matched NICs will be configured.

Reply:
when :
`s.DeviceID` is a VF id,
`iface.DeviceID` is a PF id,
`iface.NumVfs` is zero, (not configured with any VF yet)
Then:
Will the `iface` (PF) be ignored in merged config spec?

Reply:
Correct.

Reply:
In such case, does it mean PF will not be configured even numvfs is specified in node policy because PF is not listed in config spec?

Reply:
The config in spec of NodeState is the output of policies. So if the numVfs is specified, it will compare the device ID with VF's, if match, then generate the config for the PF. Otherwise, it will compare the device ID with PF's.

Reply:
Yes, it's a problem. I've updated the code. Please take a look.